numpy/core/_methods.py

```diff
@@ -187,16 +194,49 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     else:
         arrmean = arrmean.dtype.type(arrmean / rcount)
 
-    # Compute sum of squared deviations from mean
-    # Note that x may not be inexact and that we need it to be an array,
-    # not a scalar.
-    x = asanyarray(arr - arrmean)
-    if issubclass(arr.dtype.type, (nt.floating, nt.integer)):
-        x = um.multiply(x, x, out=x)
-    else:
-        x = um.multiply(x, um.conjugate(x), out=x).real
+    if issubclass(arr.dtype.type, (nt.floating, nt.complexfloating)):
+        # Compute the sum of (arr - m) and (arr - m)**2 over the last axis
+        if axis is None:
+            axis = tuple(range(arr.ndim))
+        elif isinstance(axis, int):
+            axis = (axis, )
+        else:
+            axis = tuple(sorted(axis))
+
+        dim_order = [i for i in range(arr.ndim) if i != axis[-1]] + [axis[-1]]
+        S1, S2 = umt.sumsquarediff(
+            arr.transpose(*dim_order),
+            arrmean.transpose(*dim_order),
+            keepdims=keepdims,
+            dtype=dtype
+        )
+
+        # Sum over the remaining axes
+        if len(axis) > 1:
+            S1 = umr_sum(S1, axis=axis[:-1], keepdims=keepdims)
+            S2 = umr_sum(S2, axis=axis[:-1], out=out, keepdims=keepdims)
 
-    ret = umr_sum(x, axis, dtype, out, keepdims)
+        if keepdims:
+            S1 = S1.transpose(*dim_order)
+            S2 = S2.transpose(*dim_order)
```

Comment:
This looks wrong to me, the transpose would need to be reversed. Using the `axis` argument makes this go away anyway

Reply:
This is just a permutation of two axes it's ok